### PR TITLE
Added support for HTML block comments

### PR DESCRIPTION
**Pull request by vojtechjelinek**
Support for , /\* */ are still default but for all html files the  are used.

### CODE REVIEW

**Comment by UltCombo** [comment]
Thanks for working on this.

If I recall correctly, I've tried to do this using `cursor.getScopeDescriptor()`, because you can have other languages such as CSS and JS embedded in the HTML and it would be nice to add the correct comment syntax for the scope where the cursor is at.

The problem gets more complex when you consider a selection that crosses language boundaries (i.e. the beginning of the selection and end of the selection belong to different scopes), in this case I'd priorize the language at the beginning of the selection because the user would most likely be commenting out the token which causes the scope change.

In the end, I couldn't get it working as good as [Emmet](https://github.com/emmetio/emmet-atom)'s `toggle-comment` (which works great for HTML and CSS), so I didn't include it in this package.

I'll merge this if you can add support for HTML without breaking the comment syntax on embedded languages (specifically, code inside `<script>` and `<style>` tags).

**Comment by vojtechjelinek** [comment]
So, what do you think?

**Comment by UltCombo** [comment]
Oh sorry, I didn't see that you've added new commits. I will review it soon. `:)`

File: lib/sublime-block-comment.js [hunk 1]
```diff
@@ -11,15 +11,31 @@ export default {
     const buffer = editor.getBuffer();
 
     buffer.transact(() => {
-      for (const selection of editor.getSelectionsOrderedByBufferPosition().reverse()) {
-        const range = selection.getBufferRange();
-        const textInRange = buffer.getTextInRange(range);
-
+      for (let i in editor.getSelectionsOrderedByBufferPosition().reverse()) {
+        const selection = editor.getSelectionsOrderedByBufferPosition().reverse()[i];
```
Comment (UltCombo) [inline lib/sublime-block-comment.js]: I don't quite understand the motivation behind this change:

``` diff
-      for (const selection of editor.getSelectionsOrderedByBufferPosition().reverse()) {
+      for (let i in editor.getSelectionsOrderedByBufferPosition().reverse()) {
+        const selection = editor.getSelectionsOrderedByBufferPosition().reverse()[i];
```

They seem equivalent as far as I can see.

File: lib/sublime-block-comment.js [hunk 2]
```diff
@@ -11,15 +11,31 @@ export default {
...
+          close: '*/'
         };
+        for (let lang of langs) {
+          if (lang.indexOf('text.html') > -1) {
+            commentTokens = {
+              open:  '<!--',
+              close: '-->',
+            };
+          }
+          if ((lang.indexOf('source.js.embedded.html') > -1) ||
```
Comment (UltCombo) [inline lib/sublime-block-comment.js]: We should probably test only the beginning, so it matches `source.js`, `source.js.embedded.html`, `source.js.jsx` etc.

File: lib/sublime-block-comment.js [hunk 3]
```diff
@@ -11,15 +11,31 @@ export default {
     const buffer = editor.getBuffer();
 
     buffer.transact(() => {
-      for (const selection of editor.getSelectionsOrderedByBufferPosition().reverse()) {
-        const range = selection.getBufferRange();
-        const textInRange = buffer.getTextInRange(range);
-
+      for (let i in editor.getSelectionsOrderedByBufferPosition().reverse()) {
+        const selection = editor.getSelectionsOrderedByBufferPosition().reverse()[i];
+        const langs = editor.scopeDescriptorForBufferPosition(selection.getBufferRange().start).getScopesArray();
```
Comment (UltCombo) [inline lib/sublime-block-comment.js]: Can keep `const range = selection.getBufferRange();` above this line and use `const langs = getPositionScopes(range.start);`

File: lib/sublime-block-comment.js (hunk 2 above)
Comment (vojtechjelinek) [inline lib/sublime-block-comment.js]: So you think is better use somethink like this? (testing for embedded languages maybe in separated function)

``` javascript
if (lang[0].includes('text.html')) {
          commentTokens = {
            open:  '<!--',
            close: '-->'
          };
        }
        if (lang[1].includes('source.js.embedded.html') ||
            lang[1].includes('source.css.embedded.html')) {
              commentTokens = {
                open:  '/*',
                close: '*/'
              };
            } 
```

Comment (UltCombo) [inline lib/sublime-block-comment.js]: Yes, that can work.
I'm thinking whether we can use a more generic approach, like reverse searching (from the end to beginning of the scopes array) for a scope that begins with `source.js`/`source.css`.
I'll try to implement it and see if it works.
Perhaps I'm overcomplicating things.

**Comment by UltCombo** [comment]
Looks good, I'll make a few tweaks and merge. :smiley:

**Comment by UltCombo** [comment]
Merged, thanks!
I've implemented the [approach](https://github.com/UltCombo/atom-sublime-block-comment/pull/1#discussion_r40512351) I've described above in this [commit](https://github.com/UltCombo/atom-sublime-block-comment/commit/5f1e8985f7e640ffcf3bd622f8bbe05b3a14c7be).
It seems to work fine in my initial tests, but it would be nice if you could test/review it as well.

**Comment by UltCombo** [comment]
Released as v0.2.0.

**Pull request merged**